server/app/survey.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-user-survey/server/model"
+)
+
+func (a *UserSurveyApp) SaveSurvey(survey *model.Survey) error {
+	survey.SetDefaults()
+	if err := survey.IsValid(); err != nil {
+		return errors.Wrap(err, "SaveSurvey: survey is not valid")
+	}
+
+	return a.store.SaveSurvey(survey)
+}
+
+func (a *UserSurveyApp) GetInProgressSurvey() (*model.Survey, error) {
+	surveys, err := a.store.GetSurveysByStatus(model.SurveyStatusInProgress)
+	if err != nil {
+		return nil, errors.Wrap(err, "GetInProgressSurvey: failed to get in progress surveys from database")
+	}
+
+	if len(surveys) > 1 {
+		var surveyIDs string
+		for _, survey := range surveys {
+			surveyIDs += " " + survey.ID
+		}
+
+		return nil, fmt.Errorf("more than one in-progress survey found in the database. There should only be one in-progress survey, in_progress_survey_count: %d, surveyIDs: %s", len(surveys), surveyIDs)
+	}
```

Comment:
@harshilsharma63 Not a security comment, but considering the checks we perfrom in `JobManageSurveyStatus` before we check for a new survey to start, how can we end up with more than one survey in progress?

Reply:
Plus, if we don't allow two concurrent surveys, shouldn't we check for it before we save a new survey?

Reply:
There is no code path that would allow more than one in-progress survey. This check is just here because I'm receiving an array of surveys and thought why not check for it anyways. Its especially important in the initial days of this plugin while its still being tested.

We do ensure that an in-progress survey, if one exists, is stopped (if needed) before starting a new one (if needed).